luigi_pipeline/sex_related_checks.py

```diff
@@ -0,0 +1,568 @@
+        sex_ht = self.run_hails_impute_sex(
+            mt,
+            ref_genome,
+            self.output_directory().path,
+            xy_fstat_threshold,
+            xx_fstat_threshold,
+            aaf_threshold,
+        )
+        sex_ht = sex_ht.checkpoint(
+            f"{self.temp_directory().path}temp_sex.ht", overwrite=True
+        )
+
+        if use_y_cov:
+            final_annotations.extend(
+                [
+                    f"chr{normalization_contig}_mean_dp",
+                    "chrY_mean_dp",
+                    "normalized_y_coverage",
+                ]
+            )
+            norm_ht = self.get_chr_cov(
+                mt, ref_genome, normalization_contig, call_rate_threshold
+            )
+            sex_ht = sex_ht.annotate(**norm_ht[sex_ht.s])
+            chry_ht = self.get_chr_cov(mt, ref_genome, "Y", call_rate_threshold)
+            sex_ht = sex_ht.annotate(**chry_ht[sex_ht.s])
+            sex_ht = sex_ht.annotate(
+                normalized_y_coverage=hl.or_missing(
+                    sex_ht[f"chr{normalization_contig}_mean_dp"] > 0,
+                    sex_ht.chrY_mean_dp / sex_ht[f"chr{normalization_contig}_mean_dp"],
+                )
+            )
+            if add_x_cov:
+                final_annotations.extend(["chrX_mean_dp", "normalized_x_coverage"])
+                chrx_ht = self.get_chr_cov(mt, ref_genome, "X", call_rate_threshold)
+                sex_ht = sex_ht.annotate(**chrx_ht[sex_ht.s])
+                sex_ht = sex_ht.annotate(
+                    normalized_x_coverage=hl.or_missing(
+                        sex_ht[f"chr{normalization_contig}_mean_dp"] > 0,
+                        sex_ht.chrX_mean_dp
+                        / sex_ht[f"chr{normalization_contig}_mean_dp"],
+                    )
+                )
+            sex_ht = sex_ht.annotate(
+                ambiguous_sex=hl.is_missing(sex_ht.is_female),
+                sex_aneuploidy=(sex_ht.is_female)
+                & hl.is_defined(sex_ht.normalized_y_coverage)
+                & (sex_ht.normalized_y_coverage > y_cov_threshold)
+                | (~sex_ht.is_female)
+                & hl.is_defined(sex_ht.normalized_y_coverage)
+                & (sex_ht.normalized_y_coverage < y_cov_threshold),
+            )
+
+            sex_expr = (
+                hl.case()
+                .when(sex_ht.ambiguous_sex, "ambiguous_sex")
+                .when(sex_ht.sex_aneuploidy, "sex_aneuploidy")
+                .when(sex_ht.is_female, "XX")
+                .default("XY")
+            )
+
+        else:
+            sex_ht = sex_ht.annotate(ambiguous_sex=hl.is_missing(sex_ht.is_female))
+            sex_expr = hl.if_else(
+                sex_ht.ambiguous_sex,
+                "ambiguous_sex",
+                hl.if_else(sex_ht.is_female, "XX", "XY"),
+            )
+        sex_ht = sex_ht.annotate(sex=sex_expr)
+        sex_ht.write(self.output().path, stage_locally=True, overwrite=True)
```

Comment:
just to clarify - at this point the output of this task is a table with the imputed sex, its is not doing any validation against the expected sex?